src/v/kafka/client/retry_with_mitigation.h

```diff
@@ -37,7 +40,7 @@ auto retry_with_mitigation(
       std::move(errFunc),
       std::exception_ptr(),
       [retries, retry_base_backoff](
-        Func& func, ErrFunc& errFunc, std::exception_ptr& eptr) {
+        const Func& func, ErrFunc& errFunc, std::exception_ptr& eptr) {
```

Comment:
```
This fixes a bug where the func was being unconditionally moved
into a continuation, so lambda captures are not available for
the retry.
```

can you add some more context to this so someone in the future doesn't need to figure out why it was being moved?

Reply:
Expanded the commit message